ui/src/components/components/CustomInput.jsx

```diff
@@ -11,36 +11,33 @@ function CustomInput({
 }) {
   const { state } = useApplicationContext();
   const { assets } = state;
-  const [balance, setBalance] = useState(asset[type]?.purse?.balance);
-  const onMax = () => {
-    asset[type] &&
-      handleChange({ target: { value: asset[type].purse.balance } });
-  };
-  const purseBalance = useMemo(() => {
-    let obj = {};
-    let exit = 0;
+
+  const balance = useMemo(() => {
+    let purse;
+    let exit = false;
     assets.forEach(assetobj => {
       if (exit) return;
-      assetobj.purses.forEach(cpurse => {
+      assetobj.purses.forEach(p => {
         if (exit) return;
-        if (cpurse?.name === asset[type]?.purse?.name) {
-          obj = cpurse;
-          exit = 1;
+        if (p?.name === asset[type]?.purse?.name) {
+          purse = p;
+          exit = true;
         }
       });
     });
-    return obj.balance;
-  }, [assets]);
-  useEffect(() => {
-    asset[type]?.purse?.balance && !purseBalance
-      ? setBalance(asset[type]?.purse?.balance)
-      : setBalance(purseBalance);
-  }, [assets, asset[type]]);
+    return purse?.balance;
+  }, [assets, asset]);
+
+  const onMax = () => {
+    asset[type] && handleChange({ target: { value: balance } });
```

Comment:
I get the `&&` in JSX but let's use normal control structures in imperative code.
```suggestion
    if (asset[type]) {
      handleChange({ target: { value: balance } });
    }
```
It could also help to comment why `asset[type]` could be falsy when `onMax` is called.

Reply:
Ack, but realized this conditional isn't needed since the max button won't render without `asset[type]`